presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/AddLocalExchanges.java

```diff
@@ -350,6 +366,17 @@ public PlanWithProperties visitAggregation(AggregationNode node, StreamPreferred
             return deriveProperties(result, child.getProperties());
         }
 
+        private List<VariableReferenceExpression> extractPreGroupedSymbols(List<VariableReferenceExpression> groupingKeys, Set<VariableReferenceExpression> exludedColumns)
+        {
+            ImmutableList.Builder<VariableReferenceExpression> preGroupedSymbolsBuilder = ImmutableList.builder();
+            for (VariableReferenceExpression groupingKey : groupingKeys) {
```

Comment:
we usually prefer stream apis when we can because it's more succict:
```
return groupingKeys.stream()
            .filter(groupingKey -> !excludedColumns.contains(groupingKey))
            .collect(toImmutableList())
```